### PR TITLE
Add support for CSE Insights related resources

### PR DESCRIPTION
**Pull request by pmontiel-sumo**

### CODE REVIEW

File: sumologic/resource_sumologic_cse_insights_configuration_test.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("CSE Insights Status destruction check: CSE Insights Status ID is not set")
+		}
+
+		s, err := client.GetCSEInsightsConfiguration()
+		if err != nil {
+			return fmt.Errorf("Encountered an error: " + err.Error())
+		}
+		if s != nil {
+			if s.Threshold != 0 && s.LookbackDays != 0 {
```
Comment (josh-williams) [inline sumologic/resource_sumologic_cse_insights_configuration_test.go:50]: Once you do delete it, `Treshold`/`LookbackDays` should be nil, not 0, right?

File: website/docs/r/cse_insights_configuration.html.markdown [hunk 2]
```diff
@@ -0,0 +1,35 @@
+---
+layout: "sumologic"
+page_title: "SumoLogic: sumologic_cse_insights_configuration"
+description: |-
+  Provides a CSE Insights Configuration
+---
+
+# sumologic_cse_insights_configuration
+Provides a CSE Insights Configuration.
```
Comment (josh-williams) [inline website/docs/r/cse_insights_configuration.html.markdown:9]: We should maybe say "Provides the CSE Insights Configuration for the whole organization. There can be only one configuration per organization." or something like that.

**Review by josh-williams — COMMENTED** [review]
A couple of minor comments (and I'll still let someone else give the final approval for the next couple probably)

File: website/docs/r/cse_insights_configuration.html.markdown (hunk 2 above)
Comment (pmontiel-sumo) [reply website/docs/r/cse_insights_configuration.html.markdown:9]: done under:
https://github.com/SumoLogic/terraform-provider-sumologic/pull/274/commits/22ae5d0e51f44dc94e2ed6cb28653422ce2ef257

File: sumologic/resource_sumologic_cse_insights_configuration_test.go (hunk 1 above)
Comment (pmontiel-sumo) [reply sumologic/resource_sumologic_cse_insights_configuration_test.go]: done under
https://github.com/SumoLogic/terraform-provider-sumologic/pull/274/commits/22ae5d0e51f44dc94e2ed6cb28653422ce2ef257

File: sumologic/sumologic_cse_insights_status.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+	ID          string `json:"id,omitempty"`
+	Description string `json:"description,omitempty"`
+	Name        string `json:"name,omitempty"`
+}
+
+type CSEInsightsStatusGet struct {
+	ID          string `json:"id,omitempty"`
+	Description string `json:"description,omitempty"`
+	Name        string `json:"name,omitempty"`
+	DisplayName string `json:"displayName,omitempty"`
```
Comment (sumovishal) [inline sumologic/sumologic_cse_insights_status.go:92]: `DisplayName` is not there in the resource object. We should add it to the resource object and mark it as computed.

File: website/docs/r/cse_insights_status.html.markdown [hunk 4]
```diff
@@ -0,0 +1,36 @@
+---
+layout: "sumologic"
+page_title: "SumoLogic: sumologic_cse_insights_status"
+description: |-
+  Provides a CSE Insights Status
```
Comment (sumovishal) [inline website/docs/r/cse_insights_status.html.markdown:5]: "CSE" -> "Sumo Logic CSE"

File: website/docs/r/cse_insights_status.html.markdown [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+---
+
+# sumologic_cse_insights_status
+Provides a CSE Insights Status.
```
Comment (sumovishal) [inline website/docs/r/cse_insights_status.html.markdown:9]: "CSE" -> "Sumo Logic CSE"

Comment (sumovishal) [reply website/docs/r/cse_insights_status.html.markdown:9]: Can we add more details here?

File: website/docs/r/cse_insights_resolution.html.markdown [hunk 6]
```diff
@@ -0,0 +1,38 @@
+---
+layout: "sumologic"
+page_title: "SumoLogic: sumologic_cse_insights_resolution"
+description: |-
+  Provides a CSE Insights Resolution
+---
+
+# sumologic_cse_insights_resolution
+Provides a CSE Insights Resolution.
```
[the remark this answers is not in the document]
Comment (sumovishal) [reply website/docs/r/cse_insights_resolution.html.markdown:9]: Can we add more details here?

File: sumologic/resource_sumologic_cse_insights_configuration.go [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+
+	return resourceSumologicCSEInsightsConfigurationRead(d, meta)
+}
+
+func resourceSumologicCSEInsightsConfigurationCreate(d *schema.ResourceData, meta interface{}) error {
+	//we are not really creating new object in backend, using constant id for terraform resource
+	d.SetId(hash("cse-insights-configuration"))
+	return resourceSumologicCSEInsightsConfigurationUpdate(d, meta)
+}
+func hash(s string) string {
```
[the remark this answers is not in the document]
Comment (pmontiel-sumo) [reply sumologic/resource_sumologic_cse_insights_configuration.go]: done under https://github.com/SumoLogic/terraform-provider-sumologic/pull/274/commits/e6319aaa89138ade60db7b72b35891e1990abc70

File: website/docs/r/cse_insights_configuration.html.markdown [hunk 8]
```diff
@@ -0,0 +1,35 @@
+---
+layout: "sumologic"
+page_title: "SumoLogic: sumologic_cse_insights_configuration"
+description: |-
+    Provides the CSE Insights Configuration for the whole organization. There can be only one configuration per organization.
+---
+
+# sumologic_cse_insights_configuration
+Provides a CSE Insights Configuration.
+
```
[the remark this answers is not in the document]
Comment (pmontiel-sumo) [reply website/docs/r/cse_insights_configuration.html.markdown:10]: done under https://github.com/SumoLogic/terraform-provider-sumologic/pull/274/commits/e6319aaa89138ade60db7b72b35891e1990abc70

File: website/docs/r/cse_insights_configuration.html.markdown [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+  threshold = 12
+}
+```
+
+## Argument reference
+
+The following arguments are supported:
+
+- `lookback_days` - (Optional) Detection window expressed in days.
+- `threshold` - (Optional) Detection threshold.
```
[the remark this answers is not in the document]
Comment (pmontiel-sumo) [reply website/docs/r/cse_insights_configuration.html.markdown]: done under https://github.com/SumoLogic/terraform-provider-sumologic/pull/274/commits/e6319aaa89138ade60db7b72b35891e1990abc70

File: website/docs/r/cse_insights_configuration.html.markdown [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+
+The following attributes are exported:
+
+- `id` - The internal ID of the insights configuration.
+
+## Import
+
+Insights Configuration can be imported using the field id, e.g.:
```
[the remark this answers is not in the document]
Comment (pmontiel-sumo) [reply website/docs/r/cse_insights_configuration.html.markdown:32]: done under https://github.com/SumoLogic/terraform-provider-sumologic/pull/274/commits/e6319aaa89138ade60db7b72b35891e1990abc70

File: website/docs/r/cse_insights_configuration.html.markdown [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+```hcl
+terraform import sumologic_cse_insights_configuration.insights_configuration id
```
[the remark this answers is not in the document]
Comment (pmontiel-sumo) [reply website/docs/r/cse_insights_configuration.html.markdown]: done under https://github.com/SumoLogic/terraform-provider-sumologic/pull/274/commits/e6319aaa89138ade60db7b72b35891e1990abc70

File: sumologic/resource_sumologic_cse_insights_resolution.go [hunk 12]
```diff
@@ -0,0 +1,153 @@
...
+		if parentId == 1 {
+			parentName = "Resolved"
+		} else if parentId == 2 {
+			parentName = "False Positive"
+		} else if parentId == 3 {
+			parentName = "No Action"
+		} else if parentId == 4 {
+			parentName = "Duplicate"
+		}
+	}
```
[the remark this answers is not in the document]
Comment (pmontiel-sumo) [reply sumologic/resource_sumologic_cse_insights_resolution.go:80]: Error will occur when calling API.

File: website/docs/r/cse_insights_configuration.html.markdown [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+    Provides the CSE Insights Configuration for the whole organization. There can be only one configuration per organization.
+---
+
+# sumologic_cse_insights_configuration
+Provides a CSE Insights Configuration.
+
+## Example Usage
+```hcl
+resource "sumologic_cse_insights_configuration" "insights_configuration" {
+  lookback_days = 13
```
[the remark this answers is not in the document]
Comment (pmontiel-sumo) [reply website/docs/r/cse_insights_configuration.html.markdown]: done under https://github.com/SumoLogic/terraform-provider-sumologic/pull/274/commits/92d16072a6c9c2522bdb5a18ca1775eeafd79f54

File: website/docs/r/cse_insights_resolution.html.markdown (hunk 6 above)
[the remark this answers is not in the document]
Comment (pmontiel-sumo) [reply website/docs/r/cse_insights_resolution.html.markdown]: done under https://github.com/SumoLogic/terraform-provider-sumologic/pull/274/commits/92d16072a6c9c2522bdb5a18ca1775eeafd79f54

File: website/docs/r/cse_insights_resolution.html.markdown [hunk 14]
```diff
@@ -0,0 +1,38 @@
+---
+layout: "sumologic"
+page_title: "SumoLogic: sumologic_cse_insights_resolution"
+description: |-
+  Provides a CSE Insights Resolution
```
[the remark this answers is not in the document]
Comment (pmontiel-sumo) [reply website/docs/r/cse_insights_resolution.html.markdown]: done under https://github.com/SumoLogic/terraform-provider-sumologic/pull/274/commits/92d16072a6c9c2522bdb5a18ca1775eeafd79f54

File: website/docs/r/cse_insights_status.html.markdown (hunk 4 above)
[the remark this answers is not in the document]
Comment (pmontiel-sumo) [reply website/docs/r/cse_insights_status.html.markdown]: done under https://github.com/SumoLogic/terraform-provider-sumologic/pull/274/commits/92d16072a6c9c2522bdb5a18ca1775eeafd79f54

File: website/docs/r/cse_insights_resolution.html.markdown (hunk 14 above)
[the remark this answers is not in the document]
Comment (pmontiel-sumo) [reply website/docs/r/cse_insights_resolution.html.markdown]: done under https://github.com/SumoLogic/terraform-provider-sumologic/pull/274/commits/92d16072a6c9c2522bdb5a18ca1775eeafd79f54

File: sumologic/sumologic_cse_insights_status.go (hunk 3 above)
Comment (pmontiel-sumo) [reply sumologic/sumologic_cse_insights_status.go:92]: done under https://github.com/SumoLogic/terraform-provider-sumologic/pull/274/commits/92d16072a6c9c2522bdb5a18ca1775eeafd79f54

File: website/docs/r/cse_insights_resolution.html.markdown (hunk 6 above)
[the remark this answers is not in the document]
Comment (pmontiel-sumo) [reply website/docs/r/cse_insights_resolution.html.markdown]: done under https://github.com/SumoLogic/terraform-provider-sumologic/pull/274/commits/92d16072a6c9c2522bdb5a18ca1775eeafd79f54

File: sumologic/resource_sumologic_cse_insights_resolution.go [hunk 15]
```diff
@@ -0,0 +1,153 @@
...
+	return parentName
+}
+
+func parentNameToParentId(parentName string) int {
+
+	parentId := -1
+
+	if parentName != "" {
```
[the remark this answers is not in the document]
Comment (pmontiel-sumo) [reply sumologic/resource_sumologic_cse_insights_resolution.go]: done under https://github.com/SumoLogic/terraform-provider-sumologic/pull/274/commits/92d16072a6c9c2522bdb5a18ca1775eeafd79f54

File: website/docs/r/cse_insights_status.html.markdown (hunk 4 above)
Comment (pmontiel-sumo) [reply website/docs/r/cse_insights_status.html.markdown]: done under https://github.com/SumoLogic/terraform-provider-sumologic/pull/274/commits/92d16072a6c9c2522bdb5a18ca1775eeafd79f54

File: website/docs/r/cse_insights_status.html.markdown (hunk 5 above)
Comment (pmontiel-sumo) [reply website/docs/r/cse_insights_status.html.markdown]: done under https://github.com/SumoLogic/terraform-provider-sumologic/pull/274/commits/d6c3b409bfda3ea68dbdf9e0034757c6ae2cee19

File: website/docs/r/cse_insights_configuration.html.markdown [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+
+The following attributes are exported:
+
+- `ID` - The internal ID of the insights configuration.
+
+## Import
+
+Insights Configuration can be imported using the field id, e.g.:
+```hcl
+terraform import sumologic_cse_insights_configuration.insights_configuration ID
```
Comment (sumovishal) [inline website/docs/r/cse_insights_configuration.html.markdown:34]: Wouldn't ID always be "cse-insights-configuration"? The import section should be similar to Policies resource https://registry.terraform.io/providers/SumoLogic/sumologic/latest/docs/resources/policies#import

**Review by sumovishal — APPROVED** [review]
LGTM except one comment. Please address it before merging.

**Pull request merged**